### PR TITLE
PORTALS-2473

### PR DESCRIPTION
**Pull request by jay-hodgson**
Screen Shot 2023-01-30 at 4 21 58 PM

Screen Shot 2023-01-30 at 4 32 15 PM

### CODE REVIEW

File: src/AppInitializer.tsx [hunk 1]
```diff
@@ -163,7 +164,9 @@ function AppInitializer(props: { children?: React.ReactNode }) {
 
   useEffect(() => {
     // on first time, also check for the SSO code
-    SynapseClient.detectSSOCode()
+    SynapseClient.detectSSOCode(undefined, error => {
+      displayToast(error, 'danger')
```
Comment (jay-hodgson) [inline src/AppInitializer.tsx:168]: Surface any errors

File: src/components/AccountSettings.tsx [hunk 2]
```diff
@@ -209,10 +212,6 @@ export const AccountSettings = () => {
                       value={location}
                     />
                   </FormGroup>
-                  <FormGroup>
-                    <FormLabel>Institutional affiliation</FormLabel>
```
Comment (jay-hodgson) [inline src/components/AccountSettings.tsx:213]: Has no onChange() and is not mapped to a UserProfile field.  I think this is supposed to be the Current affiliation (UserProfile.company)

File: src/components/AccountSettings.tsx [hunk 3]
```diff
@@ -331,7 +330,26 @@ export const AccountSettings = () => {
...
+                      variant="outlined"
+                      sx={credentialButtonSX}
+                      onClick={() => setShowUnbindORCiDDialog(true)}
+                    >
+                      Unlink your ORCID profile
+                    </Button>
+                  ) : (
+                    <ORCiDButton
+                      sx={credentialButtonSX}
+                      redirectAfter={`${SynapseClient.getRootURL()}authenticated/myaccount`}
```
Comment (jay-hodgson) [inline src/components/AccountSettings.tsx:344]: Used to redirect to the default path, which was in the Profile Validation workflow.  Need to set this to go back to this account page.

File: src/components/AccountSettings.tsx [hunk 4]
```diff
@@ -331,7 +330,26 @@ export const AccountSettings = () => {
...
                     )}
                   </div>
-                  <ORCiDButton sx={credentialButtonSX} />
+                  {orcid ? (
+                    <Button
+                      variant="outlined"
+                      sx={credentialButtonSX}
+                      onClick={() => setShowUnbindORCiDDialog(true)}
+                    >
+                      Unlink your ORCID profile
```
Comment (jay-hodgson) [inline src/components/AccountSettings.tsx:339]: New Unlink button

File: src/components/ProfileValidation/UnbindORCiD.tsx [hunk 5]
```diff
@@ -1,16 +1,16 @@
-import React, { useState } from 'react'
+import React from 'react'
 import { SynapseClient, Typography } from 'synapse-react-client'
 import { useSynapseContext } from 'synapse-react-client/dist/utils/SynapseContext'
 import { PROVIDERS } from 'synapse-react-client/dist/containers/Login'
-import { Button, Modal } from 'react-bootstrap'
-import { onBindToORCiD } from './ORCiDButton'
+import { Dialog, DialogActions, DialogContent, Button } from '@mui/material'
```
Comment (jay-hodgson) [inline src/components/ProfileValidation/UnbindORCiD.tsx:5]: Moved to mui

File: src/components/ProfileValidation/VerifyIdentify.tsx [hunk 6]
```diff
@@ -51,12 +51,6 @@ export const VerifyIdentify = (props: VerifyIdentifyProps) => {
         disabled={!verificationSubmission.orcid}
       />
       <ReturnToAppButton />
-      {/* agendel todo: move the 'unlink' functionality to sage account page
```
Comment (jay-hodgson) [inline src/components/ProfileValidation/VerifyIdentify.tsx:54]: Done!

File: src/AppInitializer.tsx (hunk 1 above)
Comment (jay-hodgson) [reply src/AppInitializer.tsx:168]: Handler should be improved, this is the least I could do

File: src/components/AccountSettings.tsx [hunk 7]
```diff
@@ -331,7 +330,26 @@ export const AccountSettings = () => {
                       </span>
                     )}
                   </div>
-                  <ORCiDButton sx={credentialButtonSX} />
+                  {orcid ? (
+                    <Button
+                      variant="outlined"
+                      sx={credentialButtonSX}
```
Comment (conundrumgirl) [inline src/components/AccountSettings.tsx:336]: very nit-picky, and really just a matter of preference but  _credentialButtonSX_ is  a single prop definition {marginRight: '26px'} So I would consider moving it inline just for the sake of colocation. But then we'd have to repeat it twice for the ln336 &343 so I can see the argument for keeping it as is as well.
 May be an alternative would be to turn 
`<div className="credential-partition">` into `<Box className="credential-partition" sx={{'& button': {marginRight: '26px'}}}>` All in all I don't think it matters a whole lot one way or the other.

**Review by conundrumgirl — APPROVED** [review]

**Pull request merged**